client.go

```diff
@@ -175,32 +184,45 @@ func (cc *ClientConn) newAddrConn(addr string) *addrConn {
 // listenForConnectivityChange listens for the addrConn's connectivity to change
 // and updates the ClientConn ConnectivityStateManager.
 func (cc *ClientConn) listenForConnectivityChange() {
+	defer cc.wg.Done()
 	for {
-		s := <-cc.csCh
-
-		cc.csMgr.updateState(s)
+		select {
+		case <-cc.ctx.Done():
+			return
+		default:
+			s := <-cc.csCh
+			cc.csMgr.updateState(s)
```

Comment:
Why not select from both?
```suggestion
		case s := <-cc.csCh:
			cc.csMgr.updateState(s)
```